index.html

```diff
@@ -95,18 +103,32 @@
         background: transparent;
         z-index: 2;
       }
+
+      #loading-overlay {
+        position: absolute;
+        width: 100%;
+        top: 20%;
+        z-index: 1000;
+      }
+      #loading-overlay.hidden {
+        display: none;
+      }
      </style>
   </head>
   <body>
-    <div id="scanning-overlay">
+      <div id="loading-overlay">
+        <div>Loading...</div>
+        <p id="loading-percentage"></p>
+      </div>
+    <div id="scanning-overlay" class="hidden">
       <div class="inner">
 	      <div class="scanline"></div>
         <!-- todo: maybe instead of scanline do a fadein/fadeout of red, LOADING BAR-->
       </div>
     </div>
     <div class="footer">
       <div id="found-overlay" class="hidden"> <!-- Read Original-->
-        <div style="text-align: center; color: white; font-family: Special Elite; font-weight: 400; word-wrap: break-word;">Read original</div>
```

Comment:
moved this to global body styles, might need to tweak when testing for real, idk